sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala

```diff
@@ -266,41 +269,39 @@ object DottyPlugin extends AutoPlugin {
       },
       // ... instead, we'll fetch the compiler and its dependencies ourselves.
       scalaInstance := Def.taskDyn {
-        if (isDotty.value) Def.task {
-          val updateReport =
-            fetchArtifactsOf(
-              dependencyResolution.value,
-              scalaModuleInfo.value,
-              updateConfiguration.value,
-              (unresolvedWarningConfiguration in update).value,
-              streams.value.log,
-              scalaOrganization.value %% "dotty-doc" % scalaVersion.value)
-          val scalaLibraryJar = getJar(updateReport,
-            "org.scala-lang", "scala-library", revision = AllPassFilter)
-          val dottyLibraryJar = getJar(updateReport,
-            scalaOrganization.value, s"dotty-library_${scalaBinaryVersion.value}", scalaVersion.value)
-          val compilerJar = getJar(updateReport,
-            scalaOrganization.value, s"dotty-compiler_${scalaBinaryVersion.value}", scalaVersion.value)
-          val allJars =
-            getJars(updateReport, AllPassFilter, AllPassFilter, AllPassFilter)
-
-          makeScalaInstance(
-            state.value,
-            scalaVersion.value,
-            scalaLibraryJar,
-            dottyLibraryJar,
-            compilerJar,
-            allJars
-          )
-        }
-        else Def.task {
-          // This should really be `old` with `val old = scalaInstance.value`
-          // above, except that this would force the original definition of the
-          // `scalaInstance` task to be computed when `isDotty` is true, which
-          // would fail because `managedScalaInstance` is false.
-          Defaults.scalaInstanceTask.value
-        }
+        val isD = isDotty.value
+        val si = scalaInstance.taskValue
+        val siTaskInitialize = Def.valueStrict { si }
+        if (isD) dottyScalaInstance
```

Comment:
Can we inline some of this stuff ? e.g.
````scala
if (isDotty.value) Def.task { ... }
else Def.valueStrict { ... }
```

Reply:
I split things out so each step is easier to understand (for me). You can inline as much as you like.

Reply:
c1aec9c82d0fde63918f700b35614baa18e688ca